exporter/coralogixexporter/client.go

```diff
@@ -0,0 +1,124 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coralogixexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/coralogixexporter"
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"os"
+
+	cxpb "github.com/coralogix/opentelemetry-cx-protobuf-api/coralogixpb"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+)
+
+// CoralogixClient by Coralogix
+type CoralogixClient struct {
+	cfg      Config
+	logger   *zap.Logger
+	cfggrpc  configgrpc.GRPCClientSettings
+	conn     *grpc.ClientConn
+	client   cxpb.CollectorServiceClient
+	metadata cxpb.Metadata
+}
+
+// NewCoralogixClient by Coralogix
+func NewCoralogixClient(cfg *Config, logger *zap.Logger) *CoralogixClient {
+	// check env variables
+	endpoint := cfg.Endpoint
+	privateKey := cfg.PrivateKey
+	appName := cfg.AppName
+	subSystem := cfg.SubSystem
+
+	if os.Getenv("CORALOGIX_ENDPOINT") != "" {
+		endpoint = os.Getenv("CORALOGIX_ENDPOINT")
+	}
+	if os.Getenv("CORALOGIX_PRIVATE_KEY") != "" {
+		privateKey = os.Getenv("CORALOGIX_PRIVATE_KEY")
+	}
+	if os.Getenv("CORALOGIX_APPLICATION_NAME") != "" {
+		appName = os.Getenv("CORALOGIX_APPLICATION_NAME")
+	}
+	if os.Getenv("CORALOGIX_SUBSYSTEM_NAME") != "" {
+		subSystem = os.Getenv("CORALOGIX_SUBSYSTEM_NAME")
+	}
+
+	c := &CoralogixClient{
+		cfg:    *cfg,
+		logger: logger,
+		cfggrpc: configgrpc.GRPCClientSettings{
+			Endpoint:    endpoint,
+			Compression: "",
+			TLSSetting: configtls.TLSClientSetting{
+				TLSSetting:         configtls.TLSSetting{},
+				Insecure:           false,
+				InsecureSkipVerify: false,
+				ServerName:         "",
+			},
+			ReadBufferSize:  0,
+			WriteBufferSize: 0,
+			WaitForReady:    false,
+			Headers:         map[string]string{"ACCESS_TOKEN": privateKey, "appName": appName, "subsystemName": subSystem},
+			BalancerName:    "",
+		},
+		metadata: cxpb.Metadata{
+			ApplicationName: appName,
+			SubsystemName:   subSystem,
+		},
+	}
+	c.startConnection()
+	return c
+}
+func (c *CoralogixClient) startConnection() {
+	confTLS := &tls.Config{
+		InsecureSkipVerify: false,
+	}
+	conn, err := grpc.Dial(c.cfggrpc.Endpoint, grpc.WithTransportCredentials(credentials.NewTLS(confTLS)))
+	if err != nil {
+		c.logger.Error("connection error cannot dial to the endpoint ", zap.Error(err))
+		fmt.Print(err)
+	}
+	c.conn = conn
+	c.client = cxpb.NewCollectorServiceClient(c.conn)
+	fmt.Println("CONNECTION STATE IS :" + c.conn.GetState().String())
+	c.logger.Debug("CONNECTION STATE IS :" + c.conn.GetState().String())
+}
+
+func (c *CoralogixClient) newPost(td pdata.Traces) {
```

Comment:
This should accept a `context.Context`, and pass it as part of the outgoing context.